packages/integrations/sitemap/src/index.ts

```diff
@@ -20,7 +20,7 @@ export type SitemapOptions =
 	| {
 			filenameBase?: string;
 			filter?(page: string): boolean;
-			customPages?: string[];
+			customPages?: string[] | ((page: string) => Promise<string[]> | string[] | undefined);
```

Comment:
Can you explain why the callback allows for the return of `undefined`? If the callback's objective is to generate custom pages, I would expect it to return those pages.

Reply:
@ematipico This is to ensure backward compatibility with the previous customPages that has an array of string or undefined signature. I can remove the undefined if it's not going to be a problem

Reply:
I don't think it's the same thing.  `customPages?: string[]` is different from `customPages?: string[] | undefined`. Since we're using the former, it means that `options.customPages` might not exist, but when it exists, **its value is always an array**. The former means that its value *can be `undefined`*, which isn't our case.

This means that if a `customPages` is a function, that function must return an array, even an empty one. Do you think it makes sense?